[BUG]: API container doesn't start due to "nats: insufficient resources"

API container doesn't start with:
```
...
Signature:cebe1551ed35b140dce921e98cb291b77cdf99bb246be8600445e072bb5f453a2b21479f2551074e21da5ce8ee1d835213227f703cf09f5fe2ac3995be75a987 Claimed:false} CurrentHeight:935127 LastProcessedHeight:935127 UpgradePlan:{Name: Height:0 Binaries:map[] NodeVersion:} MLNodeKeyConfig:{WorkerPublicKey: WorkerPrivateKey:} Nats:{Host: Port:0} CurrentNodeVersion:v3.0.8 LastUsedVersion:v3.0.8 ValidationParams:{TimestampExpiration:60 TimestampAdvance:30 ExpirationBlocks:20} BandwidthParams:{EstimatedLimitsPerBlockKb:10752 KbPerInputToken:0.0023 KbPerOutputToken:0.64}}
2025/10/22 08:59:31 INFO starting nats server subsystem=Messages port=4222 host=0.0.0.0
current upgrade name v0.2.3, new upgrade name v0.2.3
current upgrade height 645399, new upgrade height 645399
panic: NATS server not ready after 3 attempts
goroutine 1 [running]:
main.main()
 /app/decentralized-api/main.go:67 +0x16a7
```

There is some interactive action:
```
Error: exit status 2
Config file /root/.dapi/api-config.yaml already exists
```

```
9:02AM INF the current symlink points to: "/root/.dapi/cosmovisor/upgrades/v0.2.3/bin/decentralized-api" module=cosmovisor
file /root/.dapi/cosmovisor/config.toml already exists, do you want to overwrite it? [y/n]: 9:02AM INF file already exists, not overriding module=cosmovisor
```
Did a full down and up.. and it's the same
```
init for nats
9:06AM INF checking on the genesis/bin directory module=cosmovisor
9:06AM INF the "/root/.dapi/cosmovisor/genesis/bin" directory already exists module=cosmovisor
9:06AM INF checking on the genesis/bin executable module=cosmovisor
9:06AM INF the "/root/.dapi/cosmovisor/genesis/bin/decentralized-api" file already exists module=cosmovisor
9:06AM INF making sure "/root/.dapi/cosmovisor/genesis/bin/decentralized-api" is executable module=cosmovisor
9:06AM INF checking on the current symlink and creating it if needed module=cosmovisor
9:06AM INF the current symlink points to: "/root/.dapi/cosmovisor/upgrades/v0.2.3/bin/decentralized-api" module=cosmovisor
file /root/.dapi/cosmovisor/config.toml already exists, do you want to overwrite it? [y/n]: 9:06AM INF file already exists, not overriding module=cosmovisor
9:06AM INF cosmovisor config.toml created at: /root/.dapi/cosmovisor/config.toml module=cosmovisor
Running decentralized-api with cosmovisor
...

2025/10/22 09:06:37 INFO starting nats server subsystem=Messages port=4222 host=0.0.0.0
current upgrade name v0.2.3, new upgrade name v0.2.3
current upgrade height 645399, new upgrade height 645399
panic: NATS server not ready after 3 attempts
```

tried
```
sudo rm .dapi/data/upgrade-info.json
```

```
source config.env && docker compose up --no-deps api --force-recreate -d
```

It seems to be progressing, however i see this in logs:
```
2025/10/22 09:10:45 ERROR tx broadcast, but failed to put in queue subsystem=Messages tx_id=c3ebe871-eb98-4bbe-9628-77147ecc0401 err="nats: insufficient resources" error-type=*nats.APIError
```

They know about 1 issue on connectivity between the two nodes we have.. will fix that

Nats is in the same container...
Then checked disk space / ram

nats storage is in `.dapi/.nats.` trying to find how to check it's size

There is no resource exhaustion on that server
```
root@CL-Gonka-NetNode:~/inference-ignite/deploy/join/.dapi/.nats# du -d1 -h .
17G ./jetstream
17G .
```

Nothing bad should happen if, when api container stopped, we just delete this dir `.dapi/.nats` and then restart

this queue is needed to maintain retry for sending transaction. by some reason it's huge

all important TXS (validation, claim reward) will still be retried

Cleared the queue, started the api container, seems to be working from what they see
```
2025/10/22 09:20:02 INFO Upgrade already ready subsystem=Upgrades name=v0.2.4
```

we should double check that in next upgrade. @gmorgachev assumption is that missed validation transactions were added many times in queue. not sure why another nodes are successfully sending them all 

it might be related to the fact that the node was offline some time ago. Those could have been old queue entries as well

This node just claimed reward for epoch 59 btw. That means no problem with sending transactions now

that part is actually also modified in new PR

--- a/decentralized-api/apiconfig/config.go
+++ b/decentralized-api/apiconfig/config.go
@@ -25,8 +25,9 @@ type Config struct {
 }
 
 type NatsServerConfig struct {
-	Host string `koanf:"host" json:"host"`
-	Port int    `koanf:"port" json:"port"`
+	Host                  string `koanf:"host" json:"host"`
+	Port                  int    `koanf:"port" json:"port"`
+	MaxMassagesAgeSeconds int64  `koanf:"max_massages_age_seconds"`
 }
 
 type UpgradePlan struct {

--- a/decentralized-api/apiconfig/config.go
+++ b/decentralized-api/apiconfig/config.go
@@ -27,7 +27,7 @@ type Config struct {
 type NatsServerConfig struct {
 	Host                  string `koanf:"host" json:"host"`
 	Port                  int    `koanf:"port" json:"port"`
-	MaxMassagesAgeSeconds int64  `koanf:"max_massages_age_seconds"`
+	MaxMessagesAgeSeconds int64  `koanf:"max_messages_age_seconds"`
 }
 
 type UpgradePlan struct {

--- a/decentralized-api/internal/nats/server/server.go
+++ b/decentralized-api/internal/nats/server/server.go
@@ -14,12 +14,11 @@ const (
 	TxsToSendStream    = "txs_to_send"
 	TxsToObserveStream = "txs_to_observe"
 
-	storageDir  = "/root/.dapi/.nats"
+	storageDir = "/root/.dapi/.nats"
+	maxAge     = 24 * 60 * 60 // 24 hours
+
 	DefaultPort = 4222
 	DefaultHost = "0.0.0.0"
-
-	maxMessages = 10000
-	maxAge      = 1 * 24 * time.Hour
 )
 
 type NatsServer interface {
@@ -44,6 +43,10 @@ func (s *server) Start() error {
 
 	if s.conf.Port == 0 {
 		s.conf.Port = DefaultPort
+	}
+
+	if s.conf.MaxMassagesAgeSeconds == 0 {
+		s.conf.MaxMassagesAgeSeconds = maxAge
 	}
 
 	logging.Info("starting nats server", types2.Messages, "port", s.conf.Port, "host", s.conf.Host)
@@ -90,8 +93,7 @@ func (s *server) createJetStreamTopics(topicNames []string) error {
 		_, err = js.AddStream(&nats.StreamConfig{
 			Name:     topic,
 			Subjects: []string{topic},
-			MaxMsgs:  maxMessages,
-			MaxAge:   maxAge,
+			MaxAge:   time.Duration(s.conf.MaxMassagesAgeSeconds) * time.Second,
 			Storage:  nats.FileStorage,
 		})
 

--- a/decentralized-api/internal/nats/server/server.go
+++ b/decentralized-api/internal/nats/server/server.go
@@ -17,6 +17,9 @@ const (
 	storageDir  = "/root/.dapi/.nats"
 	DefaultPort = 4222
 	DefaultHost = "0.0.0.0"
+
+	maxMessages = 10000
+	maxAge      = 1 * 24 * time.Hour
 )
 
 type NatsServer interface {
@@ -87,6 +90,8 @@ func (s *server) createJetStreamTopics(topicNames []string) error {
 		_, err = js.AddStream(&nats.StreamConfig{
 			Name:     topic,
 			Subjects: []string{topic},
+			MaxMsgs:  maxMessages,
+			MaxAge:   maxAge,
 			Storage:  nats.FileStorage,
 		})
 

--- a/decentralized-api/internal/nats/server/server.go
+++ b/decentralized-api/internal/nats/server/server.go
@@ -3,19 +3,20 @@ package server
 import (
 	"decentralized-api/apiconfig"
 	"decentralized-api/logging"
+	"time"
+
 	natssrv "github.com/nats-io/nats-server/v2/server"
 	"github.com/nats-io/nats.go"
 	"github.com/pkg/errors"
 	types2 "github.com/productscience/inference/x/inference/types"
-	"time"
 )
 
 const (
 	TxsToSendStream    = "txs_to_send"
 	TxsToObserveStream = "txs_to_observe"
 
-	storageDir = "/root/.dapi/.nats"
-	maxAge     = 24 * 60 * 60 // 24 hours
+	storageDir    = "/root/.dapi/.nats"
+	defaultMaxAge = 24 * 60 * 60 // 24 hours
 
 	DefaultPort = 4222
 	DefaultHost = "0.0.0.0"
@@ -45,8 +46,8 @@ func (s *server) Start() error {
 		s.conf.Port = DefaultPort
 	}
 
-	if s.conf.MaxMassagesAgeSeconds == 0 {
-		s.conf.MaxMassagesAgeSeconds = maxAge
+	if s.conf.MaxMessagesAgeSeconds == 0 {
+		s.conf.MaxMessagesAgeSeconds = defaultMaxAge
 	}
 
 	logging.Info("starting nats server", types2.Messages, "port", s.conf.Port, "host", s.conf.Host)
@@ -93,7 +94,7 @@ func (s *server) createJetStreamTopics(topicNames []string) error {
 		_, err = js.AddStream(&nats.StreamConfig{
 			Name:     topic,
 			Subjects: []string{topic},
-			MaxAge:   time.Duration(s.conf.MaxMassagesAgeSeconds) * time.Second,
+			MaxAge:   time.Duration(s.conf.MaxMessagesAgeSeconds) * time.Second,
 			Storage:  nats.FileStorage,
 		})
 

